src/index.js

```diff
@@ -51,7 +51,7 @@ const Alpine = {
 
         const observerOptions = {
             childList: true,
-            attributes: true,
+            attributeFilter: ['x-data'],
```

Comment:
We need elements with `x-data` attribute only - to initialize components.

By setting the option `attributeFilter` we're incredibly reducing complexity of the `MutationObserver` by triggering the callback only when we confident that element has `x-data` attribute.

You could say: "There are code that get elements with `x-data` attribute only:" https://github.com/alpinejs/alpine/blob/4aa31e2e4c24b4765389d3a63c461536033eb172/src/index.js#L39-L40
YES THERE IS!
But as you know JS is slow, and if we delegate this job to `MutationObserver` itself which is written in native code in all browsers (e.g. C++) then it's gonna check for that attribute's existence super-fast!

In `MutationObserver`'s callback we have nested loops `O(2*n)` - as you know they are slow too.
We should avoid executing nested loops as much as we can especially when they are useless.